smaily_for_drupal/src/Form/SmailyForDrupalAdminForm.php

```diff
@@ -0,0 +1,277 @@
+    ];
+
+    $form['smaily_container']['credentials']['smaily_domain'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Domain'),
+      '#default_value' => $config->get('smaily_domain', ''),
+      '#attributes' => [
+        'placeholder' => $this->t('For example "demo" from https://demo.sendsmaily.net/'),
+      ],
+      '#required' => TRUE,
+    ];
+
+    $form['smaily_container']['credentials']['smaily_username'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Username'),
+      '#default_value' => $config->get('smaily_username', ''),
+      '#required' => TRUE,
+    ];
+
+    $form['smaily_container']['credentials']['smaily_password'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Password'),
+      '#default_value' => $config->get('smaily_password', ''),
+      '#required' => TRUE,
+
+    ];
+
+    $form['smaily_container']['credentials']['validate_credentials'] = [
+      '#type' => 'button',
+      '#value' => $this->t('Validate credentials'),
+      '#ajax' => [
+        'event' => 'click',
+        'callback' => '::validateCredentials',
+        'wrapper' => 'smaily-wrapper',
+      ],
+      // Limit errors here, validateCredentials will highlight errors in autoresponder/title.
+      '#limit_validation_errors' => [],
+    ];
+
+    $username = $config->get('smaily_username');
+    $password = $config->get('smaily_password');
+    $domain = $config->get('smaily_domain');
+
+    // Fetch and update autoresponders every time form is built.
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL,
+      'https://' . $domain . '.sendsmaily.net/api/workflows.php?trigger_type=form_submitted'
+    );
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+    curl_setopt($ch, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);
+    curl_setopt($ch, CURLOPT_USERPWD, "$username:$password");
+    $autoresponders = json_decode(curl_exec($ch), TRUE);
+    curl_close($ch);
+
+    $autoresponder_list = [];
+    if (!empty($autoresponders)) {
+      foreach ($autoresponders as $autoresponder) {
+        if (!empty($autoresponder['id']) && !empty($autoresponder['title'])) {
+          $autoresponder_list[$autoresponder['id']] = trim($autoresponder['title']);
+        }
+      }
+    }
+
+    $form['smaily_container']['smaily_autoresponder'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Autoresponder'),
+      '#options' => $autoresponder_list,
+      '#default_value' => $config->get('smaily_autoresponder', 1),
+      '#empty_option' => $this->t('Select autoresponder'),
+    ];
+
+    $form['smaily_container']['smaily_button_title'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Subscribe button title'),
+      '#default_value' => $config->get('smaily_button_title', 'Subscribe to newsletter'),
+      '#required' => TRUE,
+    ];
+
+    $form['smaily_container']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save'),
```

Comment:
I think it is better if the button has information about what the user is saving. Please rename it to "Save settings".